crates/cargo-util-schemas/Cargo.toml

```
@@ -17,9 +17,14 @@ thiserror.workspace = true
 toml.workspace = true
 unicode-xid.workspace = true
 url.workspace = true
+schemars = { version = "0.8.21", features = ["preserve_order","semver"], optional = true }
+serde_json = { version = "1.0", optional = true }
```

In Cargo we use [workspace inheritances](https://doc.rust-lang.org/nightly/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace) as much as possible. You may need to do that as well, i.e,

```suggestion
schemars = { workspace = true, features = ["preserve_order","semver"], optional = true }
serde_json = { workspace = true, optional = true }
```

Could you rearrange dependencies in alphabetically order?
(Sorry that we don't have an automatic mechanism doing to this now 😅)

---

Don't forget to do this

---

I've rearranged them but I don't know how to use workspace for optional dependencies that are active only when the feature flag is set.

---

Only inherit the version. Everything else will work the same.

---

✅